### DISCUSSION TITLE
Sendgrid not work in vercel production environment

### DISCUSSION THREAD

**Discussion by ryanwentaoxu** [vercel/community#2055] [Help]
### Summary

sendgrid can work in local environment, while it doesn't work in production env. It shows unauthorized in console in vercel with 500 error code.

**Comment by amyegan** [comment]
Hi @ryanwentaoxu! The most common cause of an `unauthorized` error is an invalid [environment variable](https://vercel.com/docs/concepts/projects/environment-variables). The first step we should take is to double check that you have the required values set and that your token is not expired
